optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverConfig.java

```diff
@@ -354,10 +358,18 @@ public void setThreadFactoryClass(Class<? extends ThreadFactory> threadFactoryCl
         this.threadFactoryClass = threadFactoryClass;
     }
 
+    /**
+     * @deprecated This functionality will be removed in the next major version of OptaPlanner.
+     */
+    @Deprecated(/* forRemoval = true */)
     public ScanAnnotatedClassesConfig getScanAnnotatedClassesConfig() {
         return scanAnnotatedClassesConfig;
     }
 
+    /**
+     * @deprecated This functionality will be removed in the next major version of OptaPlanner.
```

Comment:
deprecated in favor of the Quarkus extension or Spring Boot starter which handle annotation scanning more efficiently.